src/sim-host/protocol/socket.js

```diff
@@ -7,6 +7,10 @@ var socket;
 module.exports.initialize = function (pluginHandlers, serviceToPluginMap) {
     socket = io();
     module.exports.socket = socket;
+    module.exports.notifyPluginsReady = function () {
+        telemetry.registerPluginServices(serviceToPluginMap);
+        socket.emit('register-simulation-host');
+    }
```

Comment:
You are right, this will make the code much more maintainable, so I've made these changes!